src/models/sssd_conf.py

```diff
@@ -1,34 +1,94 @@
-    # that cause an exception with default method
-    _parser.optionxform = str
-    with _sssd_conf_path.open() as f:
-        _parser.read_file(f)
+    _backup_original = Path(LIB_BACKUP, 'original_sssd.conf')
+    _backup_default = Path(LIB_BACKUP, 'default_sssd.conf')
+
+    @classmethod
+    def create(cls):
+        """
+        Creates sssd.conf file if not present in the system or update existing
+        sssd.conf file with values from template file provided by this package
+        and backup original file.
+        """
+        parser = ConfigParser()
+        # avoid problems with inserting some 'specific' values
+        parser.optionxform = str
+
+        # this method should be executed only once to create default config
+        # existing backup files indicates that this method was already executed
+        # in such case exception should be raised
+        if Path(cls._backup_original).is_file():
+            logger.error("Backup of original sssd.conf already exists")
+            raise FileExistsError('backup file exists')
+        if Path(cls._backup_default).is_file():
+            logger.error("Default sssd.conf backup already exists")
+            raise FileExistsError('backup file exists')
+
+        try:
+            with cls._sssd_conf_path.open() as config:
+                parser.read_file(config)
+            copy2(cls._sssd_conf_path, cls._backup_original)        # BACKUP
+        except FileNotFoundError:
+            logger.info(f"sssd.conf not present in {cls._sssd_conf_path}")
+            logger.info("Creating sssd.conf based on the template")
+        with cls._template.open() as template:
+            logger.info("Updating sssd.conf with values from the template")
+            parser.read_file(template)
+        with cls._sssd_conf_path.open("w") as config:
+            parser.write(config)
+        copy2(cls._sssd_conf_path, cls._backup_default)
 
     @classmethod
     def set(cls, key: str, value: str, section: str = None):
         """
         Set value in SSSD config file. Parameter 'section' has to be specified.
         """
-        previous = cls._parser.get(section, key, fallback=None)
-        if previous is None:
+        parser = ConfigParser()
+        parser.optionxform = str
+
+        try:
+            with cls._sssd_conf_path.open() as config:
+                parser.read_file(config)
+        except FileNotFoundError:
+            logger.info(f"sssd.conf not present in {cls._sssd_conf_path}")
+            raise
+
+        if not parser.has_section(section):
+            logger.warning(f"Section {section} not present in sssd.conf")
+            logger.warning(f"Adding section {section} to sssd.conf")
+            parser.add_section(section)
+
+        if not parser.has_option(section, key):
             previous = "Not set"
-        cls._parser.set(section, key, value)
+        else:
+            previous = parser.get(section, key, fallback=None)
```

Comment:
Check [docs](https://docs.python.org/3/library/configparser.html#fallback-values) and use this functionality, please